apispec/ext/marshmallow/swagger.py

```diff
@@ -454,10 +454,14 @@ class Meta:
     """
     Meta = getattr(schema, 'Meta', None)
     if getattr(Meta, 'fields', None) or getattr(Meta, 'additional', None):
-        warnings.warn(
-            "Only explicitly-declared fields will be included in the Schema Object. "
-            "Fields defined in Meta.fields or Meta.additional are ignored."
-        )
+        declared_fields = set(schema._declared_fields.keys())
+        if (set(getattr(Meta, 'fields', {})) > declared_fields
```

Comment:
I suggest you use `set()` instead of `{}` as the latter creates a `dict`, and you end up with `dict->set` type casting.